src/transformers/integrations.py

```diff
@@ -2,16 +2,23 @@
 import math
 import os
 
+from .utils import logging
+
+
+logger = logging.get_logger(__name__)
+
 
-# Import 3rd-party integrations first:
+# Import 3rd-party integrations before ML frameworks:
 
 try:
     # Comet needs to be imported before any ML frameworks
     import comet_ml  # noqa: F401
 
-    # XXX: there should be comet_ml.ensure_configured(), like `wandb`, for now emulate it
-    comet_ml.Experiment(project_name="ensure_configured")
-    _has_comet = True
+    if comet_ml.config.get_config("comet.api_key"):
+        _has_comet = True
+    else:
+        logger.warning("comet_ml is installed but `COMET_API_KEY` is not set.")
```

Comment:
excellent, but IMHO this shouldn't be `warning()` - it's an optional feature and it shouldn't be noisy if some other package installed `comet_ml` and the user doesn't have it configured.

Reply:
I merely duplicated the logic and level from the wandb logger (below this code). Let me know if you want to change both, or leave them as-is.

Reply:
this is a totally reasonable assumption, @dsblank. Thank you for clarifying.

@sgugger - what your thinking on that?

Reply:
I may be in minority of someone who looks at test logs, that's why I have been trying to clean up a lot of those unncessary warnings.

Reply:
moreover - if the test suite CI will have wandb/comet API configured and we want to test them on CI - then these should be fatal errors and not warnings.

Reply:
The reasoning for the Wandb warning if that if users installed it, they probably want to use it, so should be aware there is something missing. That being said, I'm not aware of people getting wandb installed as a dependency from another lib.
The same logic applies to comet_ml so the warning makes sense if someone has not finished setting the API but at the same time, I understand your point @stas00.

I don't think there is a way to satisfy everyone, so I'd err on the side of what makes life easier for beginners, which is to leave the warning. We can add an env variable to ignore one or the other integration (I think there is one for Wandb already) and test that env variable before raising the warning, so that the experienced user has a way of shutting up an integration they don't want.

How does that sound?

Reply:
That sounds reasonable to me as well. comet_ml respects COMET_MODE="DISABLED" so I could suppress the warning if that is set.

Reply:
That sounds great to me, and I'll do something similar for the Wandb side.

Reply:
All done, ready for final review. Thanks for all of the feedback!

Reply:
FWIW, I know for sure that I haven't installed `comet_pl` and yet it's there. I have recently reset my env, so it might take time before I find which package forced it on me. Normally one can find it with:
```
pip install pipdeptree
pipdeptree --reverse --packages comet_ml
```
but only after the package in question has been installed.

Reply:
I still don't understand what is the purpose of these warnings - if a user expects wandb/comet ml to work w/o the keys - it's not going to happen, so surely they can figure it out. Requiring users who don't use these modules endure warnings and forcing them to either succumb and register with those both services or seek out how to shut them down is not cool. The user who has an issue sorting out how to make them work, will face this once. The user who doesn't want to become wandb/comet_ml will have this warning happening all the time. Please re-think this.
I'd add an env var to activate the warning as a debug measure for someone who needs it once - not to turn it off.

Reply:
Weird! I can't think of any standard package that has comet_ml as a dependency. Let us know if you find it :)

Reply:
Will certainly do.

Reply:
Sitting on this some more, I am feeling I'm being difficult here, so I will let it go and trust that it'll sort itself out in a good time. Thank you for letting me share my thoughts.